verifier/verifier.go

```diff
@@ -72,17 +73,23 @@ func (v *verifier) Verify(ctx context.Context, signature []byte, opts notation.V
 	err = v.processSignature(ctx, signature, envelopeMediaType, trustPolicy, pluginConfig, outcome)
 	if err != nil {
 		outcome.Error = err
-		return ocispec.Descriptor{}, outcome, err
+		return outcome, err
 	}
 
 	payload := &envelope.Payload{}
 	err = json.Unmarshal(outcome.EnvelopeContent.Payload.Content, payload)
 	if err != nil {
 		outcome.Error = err
-		return ocispec.Descriptor{}, outcome, err
+		return outcome, err
 	}
 
-	return payload.TargetArtifact, outcome, nil
+	if !content.Equal(payload.TargetArtifact, desc) {
+		outcome.Error = errors.New("content descriptor mismatch")
+		err = outcome.Error
+		return outcome, err
+	}
+
+	return outcome, nil
```

Comment:
It can be simplified as
```suggestion
	if !content.Equal(payload.TargetArtifact, desc) {
		outcome.Error = errors.New("content descriptor mismatch")
	}

	return outcome, outcome.Error
```